base/pkg/generate.jl

```diff
@@ -2,25 +2,28 @@
 
 module Generate
 
-import ..Git, ..Read
+import ..LibGit2, ..Read, ...Pkg.PkgError
+importall ..LibGit2
 
-copyright_year() = readchomp(`date +%Y`)
-copyright_name(dir::AbstractString) = readchomp(Git.cmd(`config --get user.name`, dir=dir))
-github_user() = readchomp(ignorestatus(`git config --global --get github.user`))
+copyright_year() =  string(Dates.year(Dates.today()))
+copyright_name(repo::GitRepo) =
+    with(GitConfig, repo) do cfg
```

Comment:
I see this `with(GitConfig, repo) do cfg .... ` pattern a lot in the code.  Seems like it would be better to abstract it as a `getconfig(repo, key, default="")` function.